src/slice.c

```diff
@@ -483,14 +483,22 @@ SEXP vec_init(SEXP x, R_len_t n) {
 }
 
 // [[ register() ]]
-SEXP vctrs_init(SEXP x, SEXP n) {
-  n = PROTECT(vec_cast(n, vctrs_shared_empty_int, args_n, args_empty));
-  vec_assert_size(n, 1, args_n);
-  R_len_t n_ = r_int_get(n, 0);
-
-  SEXP out = vec_init(x, n_);
-
-  UNPROTECT(1);
+r_obj* ffi_init(r_obj* x, r_obj* ffi_n, r_obj* ffi_frame) {
+  struct r_lazy frame = { .x = ffi_frame, .env = r_null };
```

Comment:
Is there any reason you didn't just add `call = caller_env()` to `vec_init()`?

Reply:
If you were to add `call` to `vec_init()`, do you really need the r_lazy struct idea? Why doesn't just passing through the `call` work?

Reply:
Yes, I think it'd be weird to add call arguments to the vector manipulation functions. So I've only done it in vctrs operations that validate or check inputs. But this is still evolving. In particular, perhaps the tidyr and dplyr ports will show us that call arguments are needed.

Reply:
> If you were to add call to vec_init(), do you really need the r_lazy struct idea? Why doesn't just passing through the call work?

It's just for low-level functions to avoid evaluating error-related arguments unless needed.